Update loader being removed correctly on manually forced transition duration

IceFog72 opened the pull request:
Preventing user custom css (transition-duration: 0) to not load a page.

<!-- Put X in the box below to confirm -->

## Checklist:

- [x] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

IceFog72 committed:
Update loader.js

Preventing user custom css transition-duration: 0 loading error.

Cohee1207 commented on public/scripts/loader.js:
```diff
@@ -27,24 +27,36 @@ export async function hideLoader() {
+
+        // Check if transitions are disabled by comparing computed style
+        const transitionDuration = getComputedStyle(spinner[0]).transitionDuration;
+        const hasTransitions = transitionDuration !== '0s' && transitionDuration !== '0';
```
This will not catch ms or other units.

Conside the following
```suggestion
        const hasTransitions = parseInt(transitionDuration) > 0;
```

IceFog72 committed:
Update public/scripts/loader.js

const hasTransitions = parseFloat(spinnerComputedStyle.transitionDuration) > 0;

Co-authored-by: Cohee <18619528+Cohee1207@users.noreply.github.com>

IceFog72 commented:
maybe ?
```
const hasTransitions = parseFloat(spinnerComputedStyle.transitionDuration) > 0;
```

Cohee1207 commented:
Whatever, both will work


> maybe ?
> 
> ```
> const hasTransitions = parseFloat(spinnerComputedStyle.transitionDuration) > 0;
> ```

IceFog72 committed:
Update loader.js

IceFog72 committed:
Update loader.js

added Fallback to ensure cleanup happens even if transition event is missed

Cohee1207 commented on public/scripts/loader.js:
```diff
@@ -27,24 +27,50 @@ export async function hideLoader() {
+
+        let transitionTimeout;
+        if (hasTransitions) {
+            spinner.one('transitionend webkitTransitionEnd oTransitionEnd MSTransitionEnd', () => {
```
```js
Promise.race([
  new Promise((r) => setTimeout(r, 500)),
  new Promise((r) => spinner.one('transitionend webkitTransitionEnd oTransitionEnd MSTransitionEnd', r)),
]).finally(cleanup)
```

IceFog72 committed:
Update loader.js

 Cohee1207 suggestions

IceFog72 committed:
Update loader.js

IceFog72 committed:
Update loader.js

IceFog72 committed:
Update loader.js
